### PR TITLE
ColorPalette: Make selected color more obvious

### LINKED ISSUE #17332: Consider revising the color swatch states.
The current "Selected" color circle makes it really difficult to see the actual selected color. To combat this, Gutenberg adds a second, rectangular color swatch next to the panel title. 

Screen Shot 2019-09-04 at 3 02 01 PM

We should consider revising the circular swatch treatment in order to negate the need for that additional swatch. 

One quick idea would be to adopt the select + focus states [used for block styles](https://github.com/WordPress/gutenberg/pull/16545). Here's how that might look: 

Frame 5

_Comparison to current approach:_ 
Frame 6

This isn't quite as clear, so I think some more iteration is needed. Also worth considering: either way, we'll need a place to include a swatch if the user selects a custom color.

### PR DESCRIPTION
**Pull request by johnwatkins0**
## Description

This would resolve #17332. Currently when selecting a color from the ColorPalette it can be hard to tell which color you have selected:

64284216-11c04c00-cf27-11e9-8aaa-2dbbd4e3e398

This is especially true working with the Text Color format, which doesn't have the "Background Color" indicator that appears other places. 

This update adds a new `selectedIconProps` prop to the CircularOptionPicker. This allows extra props to be passed to the check icon that shows over the selected color. We then use this prop to pass down style overrides from the ColorPalette component. The style override uses the tinycolor `mostReadable` function to determine whether the icon should be white or dark gray.

## How has this been tested?

Ran Jest tests, which passed when snapshots were updated. Tested locally in the browser using several components that use the ColorPalette, including the table block and pull quote block.

## Screenshots 

After:

color-picker

## Types of changes

This is a backward-compatible improvement to the display of the color picker. 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/color-palette/index.js [hunk 1]
```diff
@@ -29,12 +30,21 @@ export default function ColorPalette( {
 			<CircularOptionPicker.Option
 				key={ color }
 				isSelected={ value === color }
+				selectedIconProps={
+					value === color
+						? {
+								fill: tinycolor
+									.mostReadable( color, [ '#111', '#fff' ] )
```
Comment (mtias) [inline packages/components/src/color-palette/index.js]: Should we use `#000` instead? cc @jasmussen

Comment (jasmussen) [reply packages/components/src/color-palette/index.js]: We should. How much more black could it be? And the answer is none. None more black.

Comment (johnwatkins0) [reply packages/components/src/color-palette/index.js]: 👍 Updated.

**Comment by karmatosed** [comment]
Looks great to me. I will remove the feedback label as this works well. Thanks for this @johnwatkins0. It will need a code review but once gets that let's merge!

**Review by youknowriad — APPROVED** [review]
selectedIconProps prop seems like a too specific prop (design prop) that is not very semantic for the component's API. That said, since this is an internal component, it's not that important especially cause I'm having trouble finding a better alternative.

**Comment by mtias** [comment]
Agree on the prop, also seems like it shouldn't be computed so high up the chain.

**Pull request merged**